src/utils/State.js

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useEffect, useContext } from 'react';
+import { saveToLocalStorage, loadFromLocalStorage } from './storage';
+import { SEARCH_VIDEO, APP_STATE, VIDEO, FAVORITE_VIDEOS, DARK_MODE } from './constants';
+import { message } from 'antd';
+
+const DarkTheme = {
+    PrimaryColor: "#494A4A",
+    SecondaryColor: "#0E65AD",
+    TextColor: "white",
+};
+
+const PrimaryTheme = {
+    PrimaryColor: "white",
+    SecondaryColor: "#C7E1F7",
+    TextColor: "grey",
+};
+
+export const StateContext = React.createContext(null);
+
+export const useSesion = () => {
+    const { Sesion } = useContext(StateContext);
+    if (!Sesion) {
+        message.error(`You can't access this page!`);
+    }
+    return Sesion;
+};
+
+export const State = ({ children }) => {
+    
+    const [Theme, setTheme] = useState(PrimaryTheme);
+    const [DarkMode, setDarkMode] = useState(loadFromLocalStorage(DARK_MODE));
+    const [Sesion, setSesion] = useState(loadFromLocalStorage(APP_STATE));
+    const [SearchVideo, setSearchVideo] = useState("Wizeline");
+    const [Video, setVideo] = useState(loadFromLocalStorage(VIDEO) ? loadFromLocalStorage(VIDEO) : {});
+    const [FavoriteVideos, setFavoriteVideos] = useState(loadFromLocalStorage(FAVORITE_VIDEOS) ? 
+        loadFromLocalStorage(FAVORITE_VIDEOS) : [] );
+
+    useEffect(() => {
+        if(DarkMode){
+            setTheme(DarkTheme);
+        } else {
+            setTheme(PrimaryTheme);
+        }
+    }, [DarkMode])
+
+    useEffect(() => {
+        saveToLocalStorage(DARK_MODE, DarkMode)
+    }, [DarkMode])
+
+    useEffect(() => {
+        saveToLocalStorage(APP_STATE, Sesion)
+    }, [Sesion])
+
+    useEffect(() => {
+        saveToLocalStorage(SEARCH_VIDEO, SearchVideo)
+    }, [SearchVideo])
+
+    useEffect(() => {
+        saveToLocalStorage(VIDEO, Video)
+    }, [Video])
+
+    useEffect(() => {
+        saveToLocalStorage(FAVORITE_VIDEOS, FavoriteVideos)
+    }, [FavoriteVideos])
+
+    return (
```

Comment:
This works really good but if your application starts to grow everything could start to be really complex to maintain, I suggest that you create different Providers depending the use case of the values that the provider is exposing, also I noticed that you are using only ```useState```, You have a really good progress now if you find some time, try to implement the same using ```useReducer``` https://reactjs.org/docs/hooks-reference.html#usereducer I know that it could be a little scary at the begging but once you learn how to use it you'll like it, you can use the action to update the local storage for example instead of adding the ```effects``` for each new variable.